bootstrapers/showcase/templates/shared/components/FormSection/index.tsx

```diff
@@ -81,146 +122,164 @@ const FormSection = ({ row, col, model, data }: FormSectionProps) => {
-                  onChange={(e) =>
-                    handleFormData(
-                      e.target.checked
-                        ? [...state[field.name], choice]
-                        : state[field.name].filter((item) => item !== choice),
-                      field.name
-                    )
-                  }
-                />
-                {choice}
-              </label>
+              <CheckBox
+                key={choice}
+                id={choice}
+                htmlFor={choice}
+                value={choice}
+                labelText={choice}
+                isChecked={formFieldsState[field.name]?.includes(choice)}
+                onChange={(isChecked) => {
+                  handleFormData(
+                    isChecked
+                      ? [...formFieldsState[field.name], choice]
+                      : formFieldsState[field.name].filter((item) => item !== choice),
+                    field.name
+                  );
+                }}
+                disabled={isSuccess}
+              />
             ))}
           </div>
+
         );
-      case 'Radio Buttons':
+      case 'Predefined List - Radio Buttons':
         return (
           <div key={field.id}>
             {field.choices.map((choice) => (
-              <label key={choice}>
-                <Input
-                  type="radio"
-                  value={choice}
-                  checked={state[field.name] === choice}
-                  onChange={() => handleFormData(choice, field.name)}
-                />
-                {choice}
-              </label>
+              <Radio
+                key={choice}
+                id={choice}
+                labelText={choice}
+                value={choice}
+                labelPosition='right'
+                isChecked={formFieldsState[field.name] === choice}
+                onChange={() => handleFormData(choice, field.name)}
+                disabled={isSuccess}
+              />
             ))}
           </div>
         );
       default:
         return null;
     }
   };
-
+  const form = (
+    <ContactForm onSubmit={(e) => handleSubmit(e)}>
+      {formData.formFields &&
+        Array.isArray(formData.formFields) &&
+        formData.formFields.filter((field) => !field.invisible && !field.required)
+          .map((field) => (
+            <div key={field.id}>
+              <label>{field.name}</label>
+              {renderField(field, formFieldsState, handleFormData, state.isSuccess)}
+            </div>
+          ))}
+      {formData.formFields &&
+        Array.isArray(formData.formFields) &&
+        formData.formFields.filter((field) => !field.invisible && field.required)
+          .map((field) => (
+            <RequiredFields key={field.id} >
+              <div>
+                <label className='required-label'>{field.name}</label>
+                {renderField(field, formFieldsState, handleFormData, state.isSuccess)}
```

Comment:
I overlooked that, but I think there is no need for two maps here, you can implement one only and add a condition inside for the className, something like next:
```suggestion
        <FormInput {...field} />
```

and you define this component before the whole component as:
```suggestion
const FormInput =({ name, state, ... etc}: FormFieldType) => {
    if(field.required){
        return (
          <RequiredFields key={field.id} >
              <div>
                <label className='required-label'>{field.name}</label>
                {renderField(field, formFieldsState, handleFormData, state.isSuccess)
              </div>
          </RequiredFields>
        )
    }
    
    return (
       <div key={field.id}>
              <label>{field.name}</label>
              {renderField(field, formFieldsState, handleFormData, state.isSuccess)}
            </div>
    )
}
```